### ISSUE TITLE
Export-DbaUser not overwritten the file 

### ISSUE THREAD

**Issue by Boyzik** [sqlcollaborative/dbatools#6710]
### Environmental information

PowerShell Version        : 5.1.18362.113
dbatools latest installed : 1.0.111
Culture of OS             : pt-PT

#### SQL Server:

```sql
/* REPLACE WITH output of @@VERSION */
```
Microsoft SQL Server 2017 Enterprise Edition 
```sql
/* REPLACE WITH output of @@LANGUAGE */
```
us_english
#### Host used

- [x] powershell.exe
- [x] ISE
- [ ] VS Code
- [x] Other (please specify) 
ADS

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [x] Still buggy in powershell.exe

#### Errors Received
No errors received.

#### Steps to Reproduce

Export-DbaUser -SqlInstance DevInst -Database someDB -FilePath "C:\Temp\test.txt"

I also tried mixing the -Append and -NoClobber parameters to false without success. The behaviour is the same.

I also tried on the actual latest version (v.1.0.115) and is the same.

#### Expected Behavior

Was expecting to have the same behaviour like the Export-DbaLogin that the file is always overwritten.

#### Actual Behavior

Execute sequentially the command above makes the code to be appending to the file  and not overwritten it as was expected..

**Comment by wsmelton** [comment]
Bit odd why this does not follow the same pattern as the login command. 

We explicitly set Append to true for some reason if you provide the FilePath:

https://github.com/sqlcollaborative/dbatools/blob/3eeb23e6cf85f49125765f17c1dbf0adc0887b54/functions/Export-DbaUser.ps1#L226-L233

**Comment by Boyzik** [comment]
Yeah it's really odd. 
Like the command says:
# Force append to have everything on same file
                $Append = $true
This is just to have all objects on same file instead of having a file per object script out. And it's true. This is happening the right way.
The problem is that the file should be overwritten every time that the command is executed.

**Comment by andreasjordan** [comment]
https://github.com/sqlcollaborative/dbatools/blob/3eeb23e6cf85f49125765f17c1dbf0adc0887b54/functions/Export-DbaUser.ps1#L226-L233
Why is this code inside the `foreach ($db in $InputObject) {` in line 196 and executed for every db?
I think it should be in the begin block starting line 155.

And there (in the begin block) should be a test for the parameter $Append and if it's false then delete the file if present.

**Comment by Boyzik** [comment]
I think that it's not necessary to test if the file exists to be deleted. I debug the function and I have a suggestion for a solution.

It must be commented the line 496 ($sql | Out-File -Encoding UTF8 -FilePath $FilePath -Append:$Append -NoClobber:$NoClobber) and replaced by "$sqlFinal = $sqlFinal + $sql"

After the line 504 it should be incorpored the code:
if ($sqlFinal){
                $sqlFinal | Out-File -Encoding UTF8 -FilePath $FilePath -NoClobber:$NoClobber
            } 

I've tested and it worked.
Hope to see comments about it.

Thanks!

**Comment by wsmelton** [comment]
It would be best that we match the same pattern used in the login command so they are consistent.

**Comment by Boyzik** [comment]
@wsmelton the way the Export-DbaLogin is written is really different from the Export-DbaUser. Just to start, the -Append parameter that is forced to true in some zones inside Export-DbaUser, inside Export-DbaLogin the value of this parameter is maintain along all code without changes to it.
The way how the code was written for this parameter inside Export-Dbauser (but it makes sense how it looks like after debugging it), is the reason for the wrong behaviour.  
I mean, I think that it's not possible to have the same pattern for both commands.

**Comment by ClaudioESSilva** [comment]
Export-DbaUser, whenever we don't specify the `-User` parameter is generating a file per user.
I haven't changed this behaviour, for now, as it will be a breaking change and requires more testing. (To be done later).

I have fixed the wrong default behaviour (reported on this issue). With this version, it will overwrite by default.
I took the opportunity and I have also added the `-Encoding` parameter because was missing. (we have it on `Export-Dba*` commands)
